### PR TITLE
(PoC) feat!: add experimental support for vue 3

### PR DESCRIPTION
**Pull request by luqven**
> This PR and branch are a proof of concept and subject to change at any time

Each commit is meant as its own isolated unit for review and has comments explaining the changes. 

There's a lot going on here since there has been a massive change to the vue global APIs, so I don't expect every commit to make perfect sense. Please feel free to ask for clarification. I've most certainly made a mistake somewhere.

## Necessary work
- [x] refactor components to remove use of vue-class-components
- [x] update ts shims to work with vue 3
- [x] update vue config to use vue 3 compat build
- [x] update bare minimum dev deps for vue 3 compatibility
- [ ] updates all dependencies (some vue-cli ones still needs attention)
- [ ] address all warnings
- [ ] fix broken cypress tests
- [ ] fix broken integration tests
- [ ] update README to indicate unstable release, meant for vue 3
- [ ] tbd

### CODE REVIEW

File: src/plugins/vue-imgix/ix-source.tsx [hunk 1]
```diff
@@ -11,39 +16,25 @@ const IxSourceProps = Vue.extend({
...
     };
 
     const childAttrs = {
       [attributeConfig.srcset]: srcset,
     };
 
-    return createElement('source', { attrs: childAttrs });
-  }
-}
+    return () => h('source',  childAttrs );
```
Comment (sherwinski) [inline src/plugins/vue-imgix/ix-source.tsx:38]:
```suggestion
    return () => h('source', childAttrs );
```

**Review by sherwinski — APPROVED** [review]
Overall LGTM 👍 
One question, is there any action we need to take for [this commit](https://github.com/imgix/vue-imgix/pull/409/commits/723d80f145a466130d6053d7015503555852bd8e) in the future?

**Comment by luqven** [comment]
> Overall LGTM 👍
> One question, is there any action we need to take for [this commit](https://github.com/imgix/vue-imgix/pull/409/commits/723d80f145a466130d6053d7015503555852bd8e) in the future?

Definitely. This needs to get tackled ASAP. Either we re-write the tests using the new builtin vue testing tools or we figure out what the render bug is. But to be more confident about this RC, this needs to be addressed.

**Comment by luqven** [comment]
Apols for all the force pushes here. Had to rebase w/ main and resolve some deps conflicts.

**Pull request merged**